src/JabbR-Core/Hubs/Chat.cs

```diff
@@ -444,20 +452,27 @@ public Task<RoomViewModel> GetRoomInfo(string roomName)
                 return null;
             }
 
-            return GetRoomInfoCore(room);
+            return await GetRoomInfoCore(room);
+            //return new Task<RoomViewModel>(() => thing);
         }
 
+        // This function doesn't behave well when you use .ToListAsync() on the messages query,
+        // but making it synchronous breaks some front end UI stuff. Leaving for now
         private async Task<RoomViewModel> GetRoomInfoCore(ChatRoom room)
         {
             var recentMessages = _recentMessageCache.GetRecentMessages(room.Name);
 
-            //// If we haven't cached enough messages just populate it now
+            // If we haven't cached enough messages just populate it now
             if (recentMessages.Count == 0)
             {
-                //var messages = await (from m in _repository.GetMessagesByRoom(room)
-                //                      orderby m.When descending
-                //                      select m).Take(50).ToListAsync();
-                var messages = new List<ChatMessage>();
+                var messages = (from m in _repository.GetMessagesByRoom(room)
+                                orderby m.When descending
+                                select m).Take(50).ToList();
+                //var messages = from m
+                //               in _repository.GetMessagesByRoom(room)
+                //               orderby m.When descending
+                //               select m;
+                //var messages = new List<ChatMessage>();
 
                 // Reverse them since we want to get them in chronological order
                 messages.Reverse();
```

Comment:
Try this (or create an issue for it) - seems like we could include this in the actual query itself above. If we need top 50 most recent - which we're successfully getting - can't we then just: .Take(50).OrderBy(o=>o.When).ToList();